analyses/molecular-subtyping-LGAT/03-subset-cnv-files-LGAT.Rmd

```diff
@@ -0,0 +1,288 @@
+    file = file.path(root_dir,"data",
+                     "gencode.v27.primary_assembly.annotation.gtf.gz"),
+    format = "gtf"
+  )
+# get exons and exon_id
+tx_exons<-GenomicFeatures::exons(txdb, 
+                                 columns = list("exon_rank"),
+                                 # genocode canonical transcript https://genome.ucsc.edu/cgi-bin/hgGene?db=hg38&hgg_gene=FGFR1
+                                 filter=list(tx_name = "ENST00000447712.6"))
+  
+```
+
+For manta SV calls that are within FGFR1 gene region
+
+```{r}
+# overlap filtered manta with exons
+overlaps_manta_fgfr_exon <- IRanges::mergeByOverlaps(overlaps_manta_fgfr$manta_gr,tx_exons) 
+
+# overlap filtered cnv with exons
+overlaps_cnv_fgfr_exon <- IRanges::mergeByOverlaps(overlaps_fgfr_cnv$cnv_fgfr_gr,tx_exons) 
+
+# plot exon rank for manta and cnv
+ggplot(rbind(data.frame("exon_number"=unlist(overlaps_manta_fgfr_exon$exon_rank),
+                  "type"="tandem_dup"),
+             data.frame("exon_number"=unlist(overlaps_cnv_fgfr_exon$exon_rank),
+                  "type"="cnv_dup"))
+       ,aes(x=exon_number,color=type))+geom_density()
+
+```
+Majority SV and CNV overlap >5 exons
+
+any SV overlap which has exon overlap for exon less than 10 
+```{r}
+overlaps_manta_fgfr_exon[all(overlaps_manta_fgfr_exon$exon_rank>10  &
+                        overlaps_manta_fgfr_exon$exon_rank<18)==TRUE,"Kids.First.Biospecimen.ID.Tumor"] %>% unique()
+
+```
+
+any CNV overlap which ha exon overlap for exon less than 10 
+```{r}
+
+overlaps_cnv_fgfr_exon[all(overlaps_cnv_fgfr_exon$exon_rank>10 & overlaps_cnv_fgfr_exon$exon_rank<18)==TRUE,"ID"] %>% unique()
+```
+
+All 4 biospecimens with tandem duplication in FGFR1 and 5 biospecimens with duplication in FGFR1 have atleast 1 SV/CNV overlap where all exons are between 10 and 18. 
+
+
+### CDKN deletion 
+
+```{r}
+
+cnv_anno_cdkn2 <- cnv_anno %>%
+  dplyr::filter(gene_symbol %in% c("CDKN2A","CDKN2B")) %>%
+  reshape::cast(biospecimen_id+ploidy~gene_symbol,value = "copy_number") %>%
+  # keep only if copy_number <=1
+  dplyr::mutate(CDKN2A_DEL= if_else(CDKN2A<=1,"Yes","No"),
+                CDKN2B_DEL= if_else(CDKN2B<=1,"Yes","No")) %>%
+  dplyr::distinct() 
+
+cnv_anno_cdkn2
+  
+
+```
+
+### Gather CNV subsets
+
+```{r}
+
+# merge CNV subtype annotation
+merge_CNV_substype <-dplyr::select(overlaps_manta_fgfr_df,biospecimen_id,FGFR_DUP_TANDEM) %>%
+  full_join(dplyr::select(overlaps_cnv_df ,biospecimen_id,FGFR_DUP)) %>%
+  full_join(dplyr::select(cnv_anno_cdkn2, biospecimen_id,CDKN2A_DEL,CDKN2B_DEL)) 
+
+# merge all lgat WGS samples
+lgat_wgs_df %>%
+  left_join(merge_CNV_substype,by=c("Kids_First_Biospecimen_ID"="biospecimen_id")) %>%
+  # change NA to value==No
+  replace_na(list(FGFR_DUP_TANDEM="No",
+                         FGFR_DUP="No",
+                         CDKN2A_DEL="No",
+                         CDKN2B_DEL ="No"
+                         ))%>%
+  write_tsv(file.path(subset_dir,"LGAT_cnv_subset.tsv"))
```

Comment:
```suggestion
LGAT_cnv_subset <- lgat_wgs_df %>%
  left_join(merge_CNV_substype,by=c("Kids_First_Biospecimen_ID"="biospecimen_id")) %>%
  # change NA to value==No
  replace_na(list(FGFR_DUP_TANDEM="No",
                         FGFR_DUP="No",
                         CDKN2A_DEL="No",
                         CDKN2B_DEL ="No"
                         ))
#check frequency - does this align with the LGG paper of 9.6%? We get ~3.3%, however, the paper also states that this deletion most frequently co-occurs with BRAF V600E, so if we have a lower prevalence of that SNV, then this could make sense. Let's circle back when we put all of the data together.
CNV_by_part <- LGAT_cnv_subset %>%
  left_join(lgat_specimens_df) %>%
  select(Kids_First_Participant_ID, CDKN2A_DEL) %>%
  distinct()

CNV_by_part %>%
  tally(CDKN2A_DEL == "Yes")/nrow(CNV_by_part)*100
 
  write_tsv(LGAT_cnv_subset, file.path(subset_dir,"LGAT_cnv_subset.tsv"))
```

I wanted to check the frequency of this CDKN2A/B deletion (A/B are co-deleted in these samples; they are right next to each other) with the reported frequency by patient to be sure we were not over-estimating with our CNV results. Feel free to update if you have a better way :)